### ISSUE TITLE
[Feature request] Save the URL of each image to a text file by making only one HTTP request (or two if the gallery has two pages, or three if it has three pages, etc)

### ISSUE THREAD

**Issue by a84r7a3rga76fg** [ccloli/E-Hentai-Downloader#303]

**Comment by ccloli** [comment]
If you want the page link, then just make sure you've set Settings -> Advanced -> Record and save gallery info as _File info.txt_, and checked includes _Page Links_, which is the default setting.

Saving image link is not available since the link will only valid for a short time, after that you'll see an error.

If you want to rename the file, for now it's not possible, but if you just want the image token, I guess you can just get the SHA-1 hash of the image file, then get the first 10 letters.

**Comment by a84r7a3rga76fg** [comment]
I don't want the image URL. I forgot to add that I want it to perform the action without wasting any GP, credits or hath, and without making too many page requests. I think it should just make one page request if all of the image URLs are in one page.

**Comment by ccloli** [comment]
So do you mean you want the page links or the image links?

- For page link: Search the source code and find `getAllPagesURLFin = true;`. There're 2 matches, one for normal page grid, and one for MPV, add the following line after both of them, to output page links and stop downloading:
  ```js
  pushDialog(pageURLsList.join('\n'));
  return;
  ```

- For image URL: There's no way to do that.

- For image thumbnail URL: The script doesn't use such information, so it's not available.

**Comment by a84r7a3rga76fg** [comment]
Preferably none of those. Does my edit not show up? What I'm really looking for is saving the page number, image token (SHA-1) and the extension of every image file, e.g. https://exhentai.org/s/d1b07750bd/3042776-1 will be saved as 0001_d1b07750bd.jpg.

If you're wondering why, it's because the latest restriction has made it impossible for anyone without GP, credit or hath to download the original images. Most people don't have enough GP, credit or hath to download a single gallery. Our only option is to use torrents, and these torrents often have unsorted image files.

**Comment by ccloli** [comment]
> Preferably none of those. Does my edit not show up?

I did saw that but don't understand, probably it's nearly 6 AM in my timezone and I need a sleep. 😴

> What I'm really looking for is saving the page number, image token (SHA-1) and the extension of every image file, e.g. https://exhentai.org/s/d1b07750bd/3042776-1 will be saved as 0001_d1b07750bd.jpg.

So do you want to rename the download file? For now it's not possible, but Soon™.

If you just want to extract them from the page link and get a plain text list for such naming (d1b07750bd/3042776-1 -> 0001_d1b07750bd.jpg), probably you can try the first option's code to get page links, and ask GPT to write an automated script for you.

However the page link only contains page number and image token, the file extension is not available. You need to extract the file extension from the thumbnail URL, but the link doesn't include the page number, and the script doesn't grab thumbnail URL actually, so you need to DIY.

An example of page grid:

![image](https://github.com/user-attachments/assets/0769659b-1a57-4236-be91-e0de0e8f5348)

> it's because the latest restriction has made it impossible for anyone without GP, credit or hath to download the original images.

It's not restricted in latest update, but in last year. The latest change is just hide image limits for normal user, all the other rules are still the same as previous 2023-08 updates (latest galleries can grab with image limits only, except peak hours and/or old galleries applied).

If what you mean is to download with torrents, then calculate the hash for each file, then compare with the image link, then that'd make sense, but for that case you may probably only need the page number and page token. I'm still quite not understand why you need the file extension since you've already got the file from torrents, and to order them it's pretty sure you need to write a script to do that, then you can just extract the file extension part from the file name.

**Comment by a84r7a3rga76fg** [comment]
>So do you want to rename the download file?

No, I want the page number, the image token and the extension of every image saved to a text file.

>I'm still quite not understand why you need the file extension since you've already got the file from torrents

Sometimes whoever creates the torrent likes to change the extension from png to jpg or jpg to jpeg. You'd be surprised how often they do it.

**Comment by ccloli** [comment]
> Sometimes whoever creates the torrent likes to change the extension from png to jpg or jpg to jpeg. You'd be surprised how often they do it.

Then I'd say I'm afraid it's not available, since to avoid the case you said, the script extract the filename from the image file request's response HTTP header, so that it'll use the original file name and correct file extension, and it's definitely costs limits or GPs.

**Comment by a84r7a3rga76fg** [comment]
It can be a separate script.

**Comment by ccloli** [comment]
> It can be a separate script.

Then I'd say do it yourself, since it's not related to the script's function, and I do really need a sleep, truly sorry for that. 🥲

**Comment by ccloli** [comment]
Just to inform if you're still working on this, since EH is migrating image system to v2, there's no way to extract that file information from image thumbnail URL, because the larger thumbnail images are spirits, too. It's already applied to newer galleries and rolling out to old galleries, so I'm afraid the only way to extract expected file name is to download original image.
